js_modules/dagster-ui/packages/ui-core/src/launchpad/ConfigFetch.tsx

```diff
@@ -0,0 +1,185 @@
+import {ApolloClient, gql} from '@apollo/client';
+
+import {
+  ConfigPartitionForAssetJobQuery,
+  ConfigPartitionForAssetJobQueryVariables,
+  ConfigPartitionSelectionQuery,
+  ConfigPartitionSelectionQueryVariables,
+} from './types/ConfigFetch.types';
+import {showCustomAlert} from '../app/CustomAlertProvider';
+import {PYTHON_ERROR_FRAGMENT} from '../app/PythonErrorFragment';
+import {PythonErrorInfo} from '../app/PythonErrorInfo';
+
+export async function fetchTagsAndConfigForJob(
+  client: ApolloClient<any>,
+  variables: ConfigPartitionSelectionQueryVariables,
+) {
+  const {data} = await client.query<
+    ConfigPartitionSelectionQuery,
+    ConfigPartitionSelectionQueryVariables
+  >({
+    query: CONFIG_PARTITION_SELECTION_QUERY,
+    variables,
+  });
```

Comment:
hmm, this should really be considered as part of the page load imo if it's part of the launchpad page. Not a blocker for this PR but I'm thinking maybe we just have some kind of wrapper around async functions

Reply:
Ahh this one might be tricky... I think for op jobs the launchpad loads and then you have to choose a partition before this config is fetched -- in some cases I think it does autopopulate (for presets maybe?) but not all the time.